apps/prairielearn/python/zygote.py

```diff
@@ -147,7 +151,9 @@ def worker_loop() -> None:
             # wait for a single line of input
             json_inp = sys.stdin.readline()
             # unpack the input line as JSON
-            inp = json.loads(json_inp, parse_int=zu.safe_parse_int)
```

Comment:
`orjson` doesn't have customizable parsers that would allow us to retain out `safe_parse_int` behavior. Is this a dealbreaker?

If this is absolutely necessary, we could instead do this in a postprocessing step where we consider every value and if it's an integer that fails our `is_int_json_serializable` check, we replace it with `float(value)`. However, a similar operation (`assert_all_integers_within_limits`) was actually a pretty large bottleneck, and it was great to be able to remove it.

Reply:
It seems like `orjson` might throw an exception if it runs into an oversized int: https://github.com/ijl/orjson?tab=readme-ov-file#int

So actually, it seems like this replicates the `assert_all_integers_within_limits` for free! You should even be able to throw it into old test cases and see if it works.

Reply:
Yeah, I'm already taking advantage of the "strict int" behavior for serialization, but it doesn't help us on deserialization where we want to coerce "too big" ints into floats.

I started thinking about doing it in the JS side, but that wouldn't help us because JS can't losslessly represent those ints, which is the root of the problem! This begs the question though: how would we ever find ourselves in a situation where we're trying to parse one of these too-large integers? The only way they could get into params and such is by coming _out_ of the Python worker, but we already forbid that, right?

I suppose these JSON objects can also be modified via CSV uploads, but those also flow through JS. Maybe we just need some extra validation there?

I should look back at our original notes and PR to see why we felt we needed this special parsing behavior in the first place. Surely we thought about all this before?

Reply:
As discussed in the dev meeting, we do need this behavior, and here's why: https://github.com/PrairieLearn/PrairieLearn/issues/8462#issuecomment-1712927506

I opened an issue to see if they'd consider making this opt-in native behavior: https://redirect.github.com/ijl/orjson/issues/473